storage/replica.go

```diff
@@ -740,29 +694,43 @@ func (r *Replica) endCmds(cmdKeys []interface{}, ba *roachpb.BatchRequest, err e
 // addAdminCmd executes the command directly. There is no interaction
 // with the command queue or the timestamp cache, as admin commands
 // are not meant to consistently access or modify the underlying data.
-// Admin commands must run on the leader replica.
-func (r *Replica) addAdminCmd(ctx context.Context, args roachpb.Request) (roachpb.Response, error) {
-	header := args.Header()
+// Admin commands must run on the leader replica. Batch support here is
+// only formal: the first request is executed and responded to, nothing else.
```

Comment:
yeah, this is just sloppy. Changed the comment and am panicking if a batch
of length != 1 is passed in.

On Wed, Sep 30, 2015 at 1:03 PM, Tamir Duberstein <notifications@github.com>
wrote:

> In storage/replica.go
> <https://github.com/cockroachdb/cockroach/pull/2730#discussion_r40820574>:
>
> > @@ -740,29 +694,43 @@ func (r *Replica) endCmds(cmdKeys []interface{}, ba *roachpb.BatchRequest, err e
> >  // addAdminCmd executes the command directly. There is no interaction
> >  // with the command queue or the timestamp cache, as admin commands
> >  // are not meant to consistently access or modify the underlying data.
> > -// Admin commands must run on the leader replica.
> > -func (r *Replica) addAdminCmd(ctx context.Context, args roachpb.Request) (roachpb.Response, error) {
> > -	header := args.Header()
> > +// Admin commands must run on the leader replica. Batch support here is
> > +// only formal: the first request is executed and responded to, nothing else.
>
> this pattern worries me
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2730/files#r40820574>.
>